sycl/doc/extensions/proposed/sycl_ext_oneapi_launch_queries.asciidoc

```diff
@@ -125,135 +124,153 @@ class kernel {
   /*return-type*/ ext_oneapi_get_info(T... args) const;
 };
 
-}
+namespace ext::oneapi::experimental::info::kernel {
+
+template <uint32_t Dimensions>
+struct max_work_item_sizes;
+
+struct max_work_group_size;
+struct max_num_work_groups;
+
+struct max_sub_group_size;
+struct num_sub_groups;
+
+} // namespace ext::oneapi::experimental::info::kernel
+
+} // namespace sycl
 ----
 
 [source,c++]
 ----
 template <typename Param, typename... T>
 /*return-type*/ ext_oneapi_get_info(T... args) const;
 ----
-_Constraints_: Available only when the types `+T...+` described by the parameter
-pack match the types defined in the table below.
-
-_Preconditions_: `Param` must be one of the `info::kernel` descriptors defined
+_Constraints_: `Param` must be one of the `info::kernel` descriptors defined
 in this extension.
 
-_Returns_: Information about the kernel that applies when the kernel is
-submitted with the configuration described by the parameter pack `+T...+`.
-The return type is defined in the table below.
+_Returns_: The result of the query specified by the combination of `Param` and
+any parameters.
 
-This extension adds several new queries to this interface, many of which
-already have equivalents in the `kernel_device_specific` or `device`
-namespaces.
+==== Querying valid launch configurations
 
-NOTE: These queries are queue- and not device-specific because it is
+This extension adds several new queries for reasoning about the set of valid
+launch configurations for a given kernel, many of which already have
+equivalents in the `kernel_device_specific` or `device` namespaces.
+
+[_Note_: These queries are queue- and not device-specific because it is
 anticipated that implementations will introduce finer-grained queue
-controls that impact the scheduling of kernels.
+controls that impact the scheduling of kernels._{endnote}_]
 
-NOTE: Allowing devices to return a value of 1 for these queries maximizes the
-chances that code written to use certain extension remains portable. However,
-the performance of kernels using only one work-group, sub-group or work-item
-may be limited on some (highly parallel) devices. If certain properties (e.g.
-forward progress guarantees, cross-work-group synchronization) are being used
-as part of a performance optimization, developers should check that the values
-returned by these queries is not 1.
+[_Note_: Allowing devices to return a value of 1 for these queries maximizes
+the chances that code written to use certain extension remains portable.
+However, the performance of kernels using only one work-group, sub-group or
+work-item may be limited on some (highly parallel) devices.
+If certain properties (e.g. forward progress guarantees, cross-work-group
+synchronization) are being used as part of a performance optimization,
+developers should check that the values returned by these queries is not
+1._{endnote}_]
 
```

Comment:
Horizontal bar here.

Reply:
Fixed in 006a1c8d40ae211a9701465a8ff2ceeff655b005.